src/lightning/fabric/fabric.py

```diff
@@ -653,16 +657,26 @@ def launch(self, function: Optional[Callable[["Fabric"], Any]] = None, *args: An
                 "This script was launched through the CLI, and processes have already been created. Calling "
                 " `.launch()` again is not allowed."
             )
-        if function is not None and not inspect.signature(function).parameters:
+        if function is not _do_nothing:
+            if not callable(function):
+                raise TypeError(
+                    f"`Fabric.launch(...)` needs to be a callable. Given {function}."
```

Comment:
```suggestion
                    f"`Fabric.launch(...)` needs to be a callable, but got {function}."
```
otherwise it is not a sentence.